src/main/java/org/dependencytrack/tasks/CloneProjectTask.java

```diff
@@ -39,11 +44,24 @@ public void inform(final Event e) {
         if (e instanceof CloneProjectEvent) {
             final CloneProjectEvent event = (CloneProjectEvent)e;
             final CloneProjectRequest request = event.getRequest();
-            LOGGER.info("Cloning project: " + request.getProject());
+            final UUID chainIdentifier = ((CloneProjectEvent) e).getChainIdentifier();
             try (QueryManager qm = new QueryManager()) {
+                WorkflowState workflowState = qm.updateStartTimeIfWorkflowStateExists(chainIdentifier, PROJECT_CLONE);
+                if (workflowState == null) {
+                    final var now = new Date();
+                    workflowState = new WorkflowState();
+                    workflowState.setStep(PROJECT_CLONE);
+                    workflowState.setStatus(PENDING);
+                    workflowState.setToken(chainIdentifier);
+                    workflowState.setStartedAt(now);
+                    workflowState.setUpdatedAt(now);
+                    qm.getPersistenceManager().makePersistent(workflowState);
+                }
+                LOGGER.info("Cloning project: " + request.getProject());
                 final Project project = qm.clone(UUID.fromString(request.getProject()),
                         request.getVersion(), request.includeTags(), request.includeProperties(),
                         request.includeComponents(), request.includeServices(), request.includeAuditHistory(), request.includeACL());
+                qm.updateWorkflowStateToComplete(workflowState);
```

Comment:
We should catch exceptions thrown by `qm.clone` and transition the workflow step to `FAILED`, otherwise it will forever remain in `PENDING` state when cloning failed.